tools/backups2datalad/asyncer.py

```diff
@@ -169,16 +163,16 @@ async def process_asset(
                     PurePosixPath(asset.path).name, asset.size, sha256_digest
                 )
                 remotes = await self.annex.get_key_remotes(key)
-                if remotes is not None:
+                if "text" not in tags_from_filename(asset.path):
```

Comment:
also wondering (although not new) how come code block below is not test covered although overall function is tested -- handling binary assets should be the most common use case -- could you please add some rudimentary binary file into the test?  also note that we use 

```
* annex.largefiles=((mimeencoding=binary)and(largerthan=0))
```
so 0-size files should go directly into git

Reply:
What exactly should I use for mimetype identification?  Some specific Python library or a subprocess invocation of `file`?

Reply:
Test added.  Note that I couldn't test an empty file, as Dandi Archive doesn't allow upload of empty files.

Reply:
Also, note that we don't have the asset downloaded at the point that we determine whether to treat it as text or binary; all we have is the filename, and downloading it just to examine its MIME type would defeat the point of the check.

Reply:
d'oh, indeed -- that is the point of this PR to avoid download (of large files) before annex'ing!  Could you please add one more check, could be just assert or ValueError that we are not treating as text any file larger than let's say 10MB?  so we do not miss something obnoxious addition which `identify` tells us is a text file.

Reply:
@yarikoptic So if something appears to be a text file and is more than 10 MB, do you want it to be treated as a binary file instead, or do you want an error?

Reply:
Let's go for error for now -- I want to see what we hit that way since then we might need to reconsider `.gitattributes` rule which does not include any size limit for text files

Reply:
Done.